src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -174,14 +174,16 @@ private[health] class HealthCheckActor(
   }
 
   /** Check if enough active and ready instances will remain if we kill 1 unhealthy instance */
-  def checkEnoughInstancesRunning(): Boolean = {
+  def checkEnoughInstancesRunning(unhealthyInstance: Instance): Boolean = {
     val instances: Seq[Instance] = instanceTracker.specInstancesSync(app.id)
     val activeInstanceIds: Set[Instance.Id] = instances.withFilter(_.isActive).map(_.instanceId)(collection.breakOut)
     val healthyInstances = healthByInstanceId.filterKeys(activeInstanceIds)
       .filterKeys(instanceId => !killingInFlight(instanceId))
+
+    val futureHealthyInstances = healthyInstances.filterKeys(instanceId => unhealthyInstance.instanceId == instanceId)
```

Comment:
You want to do the inverse: keep all keys which instance id is not the unhealthy one (`filterKeys` keeps every item which matches the predicate)

Reply:
done